### DISCUSSION TITLE
Ability to create a folder in storage api

### DISCUSSION THREAD

**Discussion by hallee9000** [supabase/supabase#8094] [Feature Requests]
I want to create an empty folder in a bucket but I can't find the `createFolder` API. Please consider it. Thanks

**Comment by bnjmnt4n** [comment]
Hey @leadream! In storage, you don't have to create folders first before adding specific files to a folder; you can just create a new object with `folder-a/folder-b/filename.pdf` and the folder structure will be automatically determined from the object name. Hence, there isn't strictly the concept of empty folders, and there is no `createFolder` API. If you'd like, you can just create an object with a name like the following: `folder-a/.emptyFolderPlaceholder`.

**Reply by hallee9000** [reply]
Thanks @bnjmnt4n In my case I have to create a folder at first. So I  finally figure out I can create an empty file like `folder-a/.emptyFolderPlaceholder`.

**Comment by rbhlrn** [comment]
> I want to create an empty folder in a bucket but I can't find the `createFolder` API. Please consider it. Thanks

easy, upload an empty file to the path with the name of the folder you want to create.

    const { data, error } = await supabase.storage.from('folder_bucket').upload('test_folder/welcome.txt', {
      cacheControl: '3600',
      upsert: false
    })

**Comment by vipulrawat** [comment]
How to create a folder inside a bucket via SQL? I am trying to add folders inside a bucket i create using `seed.sql`

**Reply by ikrsx** [reply]
In Supabase everything inside a bucket is refered to as an object which is stored inside the `storage.objects` table, which contains a foreign key to the `storage.buckets.bucket_id` feild of `storage.buckets` table whose value is usually equal to the name of the bucket you created first.

So, in order to **create/add a new folder inside this new bucket** you can do a simple insertion operation on `storage.objects` like this:
```
insert into storage.objects(bucket_id, name) values('avatars' ,'folder1/');
```
this way you can add new folders inside given `bucket_id`, in this case `avatars` of `name` `folder1` **(/ matters)**.
You can find out more about `storage.objects` or other `storage.*` tables using Supabase's table editor by selecting the `storage` schema.
Hope this helps :)
